service/src/main/java/eu/europa/ec/fisheries/uvms/movement/service/bean/MovementService.java

```diff
@@ -304,6 +305,15 @@ public GetMovementListByQueryResponse getMinimalList(MovementQuery query) throws
         }
     }
 
+    private List<Movement> findValuesForNestedQuery(List<SearchValue> searchKeyValues){
+        for(SearchValue searchValue :searchKeyValues){
+            if(SearchField.AREA_ID.equals(searchValue.getField())){
+                return dao.getMovementsByMatchingArea(searchValue.getValue());
```

Comment:
At some point there will be thousands of movements per area. This method will fetch thousands of movements in the memory, to send their ids back to the DB, to fetch a page of results.

Try expressing the area criterion as a subquery.